### ISSUE TITLE
what your icp code that you ran does

### ISSUE THREAD

**Issue by mithi** [panzershaft/Visual-servoing-of-the-Sawyer-robot#8]
https://github.com/panzershaft/Visual-servoing-of-the-Sawyer-robot/blob/a31ff9504fd0d248a62486c3788ebc921ec4ff22/src/my_pcl_tutorial/src/icp_example.cpp#L28

# I don't think you understand what your icp code that you ran does
# Why the output is a bunch of numbers 

As this is what your ICP does in case you didn't read the documentation 

> If the two PointClouds align correctly (meaning they are both the same cloud merely with some kind of rigid transformation applied to one of them) then `icp.hasConverged() = 1 (true)`. It then outputs the fitness score of the final transformation and some information about it. 

- From: http://pointclouds.org/documentation/tutorials/iterative_closest_point.php

# Maybe this is what you want to do 
- http://pointclouds.org/documentation/tutorials/interactive_icp.php
> This tutorial will teach you how to write an interactive ICP viewer. The program will load a point cloud and apply a rigid transformation on it. After that the ICP algorithm will align the transformed point cloud with the original. Each time the user presses “space” an ICP iteration is done and the viewer is refreshed.

Here is a visualization. 
![Visualization](http://pointclouds.org/documentation/tutorials/_images/animation.gif)
- http://pointclouds.org/documentation/tutorials/interactive_icp.php#the-explanations

# Another thing
`cloud_out` probably shouldn't be the same pcd file as `cloud_in`
Because as said in the tutorial you are merely checking if one point cloud is a rigid transformation of another. 

```
*cloud_out = *cloud_in;
```
# IN SHORT
- The output that you get is exactly what your code intended to go since your `cloud_out` is the same as your `cloud_in`  that's why it outputs `icp.hasConverged() = 1 (true)` meaning, yes, the point cloud you have loaded are exactly the same. Maybe what you actually want to do is explained in this tutorial: http://pointclouds.org/documentation/tutorials/interactive_icp.php

**Comment by mithi** [comment]
@panzershaft there is a problem with mentorcruise as I can't send messages directly to you again. Please read my message above.

**Comment by mithi** [comment]
Remember that 
from here:  http://pointclouds.org/documentation/tutorials/iterative_closest_point.php
```
pcl::PointCloud<pcl::PointXYZ>::Ptr cloud_in (new pcl::PointCloud<pcl::PointXYZ>);
```

Then you did this: 

```
if(pcl::io::loadPCDFile ("/home/soham/wow/src/my_pcl_tutorial/src/1539950567.004.pcd", *cloud_in)==-1)
	{
		PCL_ERROR ("Couldn't read first file! \n");
		return (-1);
}
```
To load the PCD file. 
- https://github.com/panzershaft/Visual-servoing-of-the-Sawyer-robot/blob/master/src/my_pcl_tutorial/src/icp_example.cpp

Which means it is now loaded in memory as a 
```
pcl::PointCloud<pcl::PointXYZ>::Ptr
```
format.  This lines from http://pointclouds.org/documentation/tutorials/interactive_icp.php
```
typedef pcl::PointXYZ PointT;
PointCloudT::Ptr cloud_in (new PointCloudT);  // Original point cloud

  if (pcl::io::loadPLYFile (argv[1], *cloud_in) < 0)
  {
    PCL_ERROR ("Error loading cloud %s.\n", argv[1]);
    return (-1);
  }
```

Show that the pcd and ply format are loaded as the same type of object in memory.
